src/client/nav/request.js

```diff
@@ -531,18 +529,22 @@ spf.nav.request.getCacheObject_ = function(cacheKey, opt_current) {
   }
   keys.push(cacheKey);
 
-  for (var i = 0, l = keys.length; i < l; i++) {
-    var cached = spf.cache.get(keys[i]);
+  var cacheValue = null;
 
-    if (cached) {
-      return {
-        key: keys[i],
-        response: cached['response'],
-        type: cached['type']
+  // Find the first cached object and break loop early when found.
+  spf.array.some(keys, function(key) {
```

Comment:
This is a fairly common pattern. It's even described in the MDN docs.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach